### PR TITLE
Feat/15631505 sidebar action

### PR DESCRIPTION
**Pull request by CzarekDryl**
https://tw.auroracreation.com/app/tasks/15631505

image
image

### CODE REVIEW

**Review by krawczykkatarzyna — APPROVED** [review]

File: src/components/v5/common/ActionSidebar/hooks.tsx [hunk 1]
```diff
@@ -388,9 +389,15 @@ export const useSidebarActionForm = () => {
   };
 };
 
-export const useActionFormProps = () => {
+export const useActionFormProps = (
+  defaultMotionValues: ActionFormProps<any>['defaultValues'],
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/hooks.tsx:393]: It should be actionDefaultValues or just defaultValues

File: src/components/v5/common/ActionSidebar/hooks.tsx [hunk 2]
```diff
@@ -565,3 +580,140 @@ export const useActionDescriptionMetadata = () => {
...
+          amount: {
+            amount: action?.amount,
+            tokenAddress: action.token?.tokenAddress,
+          },
+          from: action.fromDomain?.nativeId.toString(),
+          recipient: action.recipientAddress,
+          ...repeatableFields,
+        };
+      case ColonyActionType.MultiplePaymentMotion:
+        return {
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/hooks.tsx:624]: const [firstPayment, ...payments] = action.payments;

File: src/components/v5/common/ActionSidebar/hooks.tsx [hunk 3]
```diff
@@ -565,3 +578,157 @@ export const useActionDescriptionMetadata = () => {
...
+            (addedToken) => ({
+              token: addedToken,
+            }),
+          );
+        const colonyTokens = action.colony.tokens?.items?.map(
+          (colonyToken) => ({
+            token: colonyToken?.token?.tokenAddress,
+          }),
+        );
+        const allTokens = [...(colonyTokens || []), ...(modifiedTokens || [])];
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/hooks.tsx:683]: New line

File: src/components/v5/common/ActionSidebar/hooks.tsx [hunk 4]
```diff
@@ -565,3 +578,157 @@ export const useActionDescriptionMetadata = () => {
...
+        if (modifiedTokens && modifiedTokens?.length > 0) {
+          return {
+            [ACTION_TYPE_FIELD_NAME]: ACTION.MANAGE_TOKENS,
+            selectedTokenAddresses: allTokens,
+            ...repeatableFields,
+          };
+        }
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/hooks.tsx:690]: New line

File: src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/ActionSidebarContent.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+              heading-3 mb-2
+              md:hover:text-blue-400 md:hover:placeholder:text-blue-400 text-gray-900
+              transition-colors
+            `}
+            mode="secondary"
+            message={false}
+          />
+        )}
+        <p className="text-gray-600 font-medium">{descriptionMetadata}</p>
+        {/* @todo: add preview mode to the form */}
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/ActionSidebarContent.tsx:64]: I think you can remove this now

File: src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/ActionSidebarContent.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        <ActionTypeSelect className="mt-7 mb-3" />
+        {/* @todo: add motion action type to each action */}
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/ActionSidebarContent.tsx:66]: I think this is done as well

File: src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/ActionSidebarContent.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+          className="flex flex-col h-full"
+          ref={formRef}
+        >
+          <ActionSidebarFormContent
+            getFormOptions={getFormOptions}
+            isMotion={!!transactionId}
+          />
+        </ActionForm>
+      </div>
+      {transactionId && (
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/ActionSidebarContent.tsx:120]: Shouldn't this be displayed only when action is motion?

File: src/components/v5/common/Fields/CardSelect/CardSelect.tsx [hunk 8]
```diff
@@ -84,78 +87,86 @@ function CardSelect<TValue = string>({
...
-              portalElementRef.current = ref;
-            }}
-            className={clsx(cardClassName, 'p-6 absolute z-[60]')}
-            hasShadow
-            rounded="s"
+      {readonly ? (
+        <span className="capitalize text-md text-gray-900">
+          {selectedOption?.label}
+        </span>
+      ) : (
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/Fields/CardSelect/CardSelect.tsx:94]: This may not be correct, what if someone passes `renderSelectedValue` prop?

**Review by TomaszChmielPagepro — APPROVED** [review]

**Pull request merged**